### PR TITLE
Added tests that run on CPU

### LINKED ISSUE #4: Unit tests with pytest
Depends on #3 

Start with adding some unit tests. I'm sure given the size of the project this will be an ongoing issue for a while.

### LINKED ISSUE #5: Integration tests
Depends on #3 

We will add integration tests, perhaps some example jupyter notebooks to make sure that the project as a whole works well.

### PR DESCRIPTION
**Pull request by torrewk**
Three new tests are added into the set of 'physical unit tests'. These are lighter tests that can be run on CPU in minutes. Additionally, a batch file ('run_tests_cpu.sh') that runs only these 3 tests, specifically on CPU, is now present in the project folder. Fixes #4 Fixes #5

### CODE REVIEW

**Comment by qubixes** [comment]
The first test fails for me with a small error, the other two pass. Any immediate thoughts on this?

File: run_tests_cpu.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Force JAX to use only the CPU
+export JAX_PLATFORMS=cpu
+
+# Run pytest on the jfsd package
+pytest tests/test_class.py::TestClassCPU
```
Comment (qubixes) [inline run_tests_cpu.sh:7]: Does the following work? In that case you don't need to export the variable.
```suggestion
JAX_PLATFORMS=cpu pytest tests/test_class.py::TestClassCPU
```

Comment (torrewk) [reply run_tests_cpu.sh]: Done.

File: tests/test_class.py [hunk 2]
```diff
@@ -147,7 +146,91 @@ def test_thermal_realspace(self):
...
+    def test_thermal_realspace(self):
+        """Physical unit test for non-deterministic part of hydrodynamic calculations, that runs on CPU. 
+        
+        Compare square root of resistance/mobility obtained using Lanczsos decomposition, with the square roots computed using scipy. 
+        This test probes lubrication and real-space far-field calculation of thermal motion.
+
+        """
+        N = 25
+        dr=2.0005
+        _, _, _, testresults = main.main(
```
[the remark this answers is not in the document]
Comment (torrewk) [reply tests/test_class.py]: Tests are separated now.

File: tests/test_class.py [hunk 3]
```diff
@@ -147,7 +146,91 @@ def test_thermal_realspace(self):
...
+    def test_shear(self):
+        """Physical unit test for deterministic shear part of hydrodynamic calculations, that runs on CPU. 
+        
+        Test instantaneous response to shear of three spheres against reference vaues.
+
+        """
+        deltas = np.array([0.0001,0.001,0.01,0.1,1,10])
+        all_traj = np.zeros((len(deltas),3,3))
+        reference_traj = np.load('files/sheared_triplet_reference.npy') #load reference trajectory
+        for i in range(len(deltas)):
```
[the remark this answers is not in the document]
Comment (torrewk) [reply tests/test_class.py]: Tests are now parametrized, when possible.

**Pull request merged**